storybookjs/storybook
# CLI: Fix regex used in upgrade command

Closes #





## What I did

When running `npx storybook@next upgrade --prerelease`, the `reject` flag passed onto `npm-check-updates` was incorrect, resulting in an unsuccessful upgrade. However, the code didn't handle the error state, and therefore the upgrade command would continue:

<img src="https://github.com/storybookjs/storybook/assets/1671563/5a44a1df-2dc6-43fa-a8f4-c63d50022d58" width="550" />

This PR both fixes the regex and adds categorized errors to the upgrade CLI command:

<img src="https://github.com/storybookjs/storybook/assets/1671563/5b9b1a73-dbd6-4e3e-9879-8cd1996ab9fe" width="550" />

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

No need, it's been thoroughly tested.

1. Open a Storybook project
2. run `path/to/cli/bin/index.js upgrade --prerelease`
3. it should succeed

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
